### PR TITLE
[vivaleva] change hardcord to api

### PR DESCRIPTION
**Pull request by geunwoobaek**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    TypeError: Cannot read properties of undefined (reading 'map')
    Truncated error stack:
    
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/vivaleva/index.js:33:34)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:148:9
        at async Promise.all (index 0)
        at async Promise.all (index 1)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:193:3

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    Error: Multicall is missing a target
    Truncated error stack:
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/abi/index.js:256:35)
        at new Promise (<anonymous>)

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    Error: Multicall is missing a target
    Truncated error stack:
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/abi/index.js:256:35)
        at new Promise (<anonymous>)

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    Error: Multicall is missing a target
    Truncated error stack:
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/abi/index.js:256:35)
        at new Promise (<anonymous>)

**Comment by llamatester** [comment]
The adapter at projects/vivaleva exports TVL: 
        
 
 
    era                       138.01 k
    
    total                    138.01 k

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    Error: Multicall is missing a target
    Truncated error stack:
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/abi/index.js:256:35)
        at new Promise (<anonymous>)

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    Error: Multicall is missing a target
    Truncated error stack:
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/abi/index.js:256:35)
        at new Promise (<anonymous>)

**Comment by llamatester** [comment]
Error while running adapter at projects/vivaleva: 
        
 
  
    
    
    Error: Multicall is missing a target
    Truncated error stack:
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/abi/index.js:256:35)
        at new Promise (<anonymous>)

**Comment by llamatester** [comment]
The adapter at projects/vivaleva exports TVL: 
        
 
 
    era                       137.98 k
    
    total                    137.98 k

**Comment by llamatester** [comment]
The adapter at projects/vivaleva exports TVL: 
        
 
 
    era                       150.24 k
    
    total                    150.24 k

File: projects/vivaleva/index.js [hunk 1]
```diff
@@ -62,19 +26,26 @@ module.exports = {
...
+          abi: "uint256:totalSupply",
+          calls: syncswapWorkers.map((v) => v.lpTokenAddress),
+        }),
       ]);
 
       syncswapWorkers.forEach((w, i) => {
-        const token0 = token0s[i]
-        const token1 = token1s[i]
+        const [token0, token1] =
+          w.baseTokenAddress.toLowerCase() < w.farmingTokenAddress.toLowerCase()
```
Comment (g1nt0ki) [inline projects/vivaleva/index.js:46]: always confused, how does this logic work? isnt is safer to fetch token info from lp?

**Pull request closed without merging**